x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -1345,6 +1347,29 @@ public void testMultiMatchOptionsPushDown() {
         assertThat(expectedQuery.toString(), is(planStr.get()));
     }
 
+    public void testKnnOptionsPushDown() {
```

Comment:
I am not sure what this test is about, especially since I did not see any optimizer related changes. Maybe adding a comment on what is tested would be great!!!